server/src/main/java/org/opensearch/transport/InboundHandler.java

```diff
@@ -96,13 +66,19 @@ public class InboundHandler {
         Tracer tracer
     ) {
         this.threadPool = threadPool;
-        this.outboundHandler = outboundHandler;
-        this.namedWriteableRegistry = namedWriteableRegistry;
-        this.handshaker = handshaker;
-        this.keepAlive = keepAlive;
-        this.requestHandlers = requestHandlers;
-        this.responseHandlers = responseHandlers;
-        this.tracer = tracer;
+        this.protocolMessageHandlers = Map.of(
```

Comment:
I think we have some sort of duplication which we could live with (or address later):
 - the `InboundHandler` is called from the `InboundPipeline`
 - `InboundPipeline` is already know what the protocol it deals with so I could use the protocol specific handler
 - but the `InboundHandler` still tries to find handler based on protocol of the message 

It does not look like a big issue at the moment, @dblock wdyt?

Reply:
I'm ok with an issue to deal with this later.